### PR TITLE
Form error handling

### PR DESCRIPTION
**Pull request by fbrcsftnr**
- generic ErrorText component
- update BigInput to handle errors
- hook for user info form logic
- fix ugly collection naming form

### CODE REVIEW

File: src/components/Profile/useUserInfoForm.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    setGeneralError('');
+
+    //------------ client-side checks ------------
+    if (!username.length) {
+      setUsernameError('Required.');
+      return;
+    }
+
+    if (username.length < 2 || username.length > 20) {
+      setUsernameError('Username must be between 2 and 20 characters.');
```
Comment (kaitoo1) [inline src/components/Profile/useUserInfoForm.ts:33]: Maybe we can put these error messages in a single const like a map

File: src/components/core/BigInput/BigInput.tsx [hunk 2]
```diff
@@ -1,40 +1,52 @@
...
   );
 }
 
 const StyledBigInput = styled.input`
   padding: 0 2px;
   border: 0;
-  font-size: 32px;
-  font-family: 'Times New Roman';
-  line-height: 44px;
+  font-size: 40px;
```
Comment (kaitoo1) [inline src/components/core/BigInput/BigInput.tsx:47]: I changed these to font-size: 32px and line-height: 44px yesterday based on Jon's feedback in Markup, but figma says 40px and 48px 🤔
https://app.markup.io/markup/9170d884-2ef5-4b68-9b43-203780ceac06

Comment (kaitoo1) [reply src/components/core/BigInput/BigInput.tsx:47]: gonna ask in discord

**Review by kaitoo1 — APPROVED** [review]

File: src/components/Profile/useUserInfoForm.ts (hunk 1 above)
Comment (Robinnnnn) [reply src/components/Profile/useUserInfoForm.ts:33]: Agreed. In an ideal world we have a set of Validators that run against the input, and each validator returns a specific error const.

Example array of validators: [required, minLength, maxLength, ...] then run your input through them all and return first error (or if no error return true)

**Pull request merged**